### PR TITLE
Bug fix in freeze encoder

### PR DESCRIPTION
**Pull request by Edresson**

### CODE REVIEW

File: TTS/tts/models/vits.py [hunk 1]
```diff
@@ -717,7 +717,7 @@ def _freeze_layers(self):
             for param in self.text_encoder.parameters():
                 param.requires_grad = False
 
-            if hasattr(self, "emb_l"):
+            if hasattr(self, "emb_l") and self.emb_l is not None:
```
Comment (erogol) [inline TTS/tts/models/vits.py:720]: When it is defined but set None?

**Comment by erogol** [comment]
Also please add a test against the bug this PR fixes

File: TTS/tts/models/vits.py (hunk 1 above)
Comment (Edresson) [reply TTS/tts/models/vits.py:720]: In init_multilingual it is defined as None if the model is not multilingual. In fact, the hasattr(self, "emb_l") is not needed given that it is None if multilingual approach is not used.

**Comment by Edresson** [comment]
> Also please add a test against the bug this PR fixes

This error happens just when  args_model.freeze_encoder is activated and the multilingual approach is disabled. Do you want a unit test for every part frozen?  If you really think that a unitest is needed we need to think of a way to do it and do not increase a lot the workflow runtime. I don't think that is worth running training for every frozen module.

**Review by erogol — APPROVED** [review]

**Pull request merged**